src/inmanta/deploy/persistence.py

```diff
@@ -115,13 +119,14 @@ async def dryrun_update(self, env: UUID, dryrun_result: DryrunResult) -> None:
 
 class ToDbUpdateManager(StateUpdateManager):
 
-    def __init__(self, client: Client, environment: UUID) -> None:
+    def __init__(self, client: Client, environment: UUID, scheduler: "ResourceScheduler") -> None:
         self.environment = environment
         # TODO: The client is only here temporarily while we fix the dryrun_update
         self.client = client
         # FIXME: We may want to move the writing of the log to the scheduler side as well,
         #  when all uses of this logger are moved
         self._resource_action_logger = logging.getLogger(const.NAME_RESOURCE_ACTION_LOGGER)
+        self.scheduler = scheduler
```

Comment:
This will cause a circular import that is why I did the `TYPE_CHECKING` check. If you have another suggestion, let me know.

Reply:
I feel like in this case it might actually point at a discouraged module structure, but I'll know more once I've gone through the rest of the changes.